hazelcast/src/main/java/com/hazelcast/internal/affinity/ThreadAffinityInterceptor.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.affinity;
+
+public interface ThreadAffinityInterceptor {
```

Comment:
We don't need this interface IMHO. Too much abstraction for no good reason.

Reply:
We have the NOOP interceptor, thus the interface.

Reply:
Just add an isEnabled method to the threadAffinity class.

Reply:
Done